src/core/server/core_app/bundle_routes/fs.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+// can't use fs/promises when working with streams using file descriptors
+// see https://github.com/nodejs/node/issues/35862
+
+import Fs from 'fs';
+import { promisify } from 'util';
```

Comment:
Spent one hour trying to properly use `fs/promises` with `createReadStream`, then I had to fall back to `promisify` as it was in legacy.

TIL: promisified fs are not the exact same as the new APIs provided by `fs/promises`.